asapdiscovery-ml/asapdiscovery/ml/weights/weights.py

```diff
@@ -0,0 +1,140 @@
+import logging
+from pathlib import Path
+from typing import List, Union  # noqa: F401
+
+import requests
+import validators
+import yaml
+
+
+def fetch_weights_from_spec(
+    yamlfile: str,
+    models: Union[list[str], str],
+    local_path: str = "./_weights/",
+    force_fetch: bool = False,
+):
+    """Fetch weights from yaml spec file.
+
+    Parameters
+    ----------
+    yamlfile : str
+        Path to yaml spec file.
+    models : List[str]
+        Model names to fetch weights for.
+    local_path : str, optional
+        Local path to save weights if a remote url is provided. or to check if weights exist locally, by default "./_weights/"
+    force_fetch : bool, optional
+        Force fetch weights from remote, by default False
+
+    Raises
+    ------
+    FileNotFoundError
+        If YAML spec file does not exist.
+
+    Returns
+    -------
+    Dict of model names and paths to weights files.
+    """
+    if not Path(yamlfile).exists():
+        raise FileNotFoundError(f"Yaml spec file {yamlfile} does not exist")
+
+    with open(yamlfile) as f:
+        spec = yaml.safe_load(f)
+
+    if isinstance(models, str):
+        models = [models]
+
+    weights_files = {}
+    filename_set = set()
+    for model in models:
+        model_spec = spec[model]
+        weights = model_spec["weights"]
+        weights_files[model] = fetch_weights(weights, local_path, force_fetch)
+        if weights in filename_set:
+            raise ValueError(
+                f"Duplicate file {weights} in spec file. Please specify a unique filename for each model."
+            )
+        filename_set.add(weights)
+    return weights_files
+
+
+def fetch_weights(weights: str, path: str = None, force_fetch: bool = False) -> Path:
+    """Fetch weights from remote and save to local path.
+
+    Parameters
+    ----------
+    weights : str
+        Remote url or local path to weights.
+    path : str, optional
+        Local path to save weights if a remote url is provided. or to check if weights exist locally, by default None
+    force_fetch : bool, optional
```

Comment:
same thing as before
```suggestion
    force_fetch : bool, default=False
```